src/passes/ifStatementTempVarPostpender.ts

```diff
@@ -0,0 +1,200 @@
+      'An', // Most of the functions this pass removes are generated by the IfFunctionaliser
+    ]);
+    passKeys.forEach((key) => this.addPassPrerequisite(key));
+  }
+
+  visitCairoFunctionDefinition(node: CairoFunctionDefinition, ast: AST): void {
+    this.liveVars = new Set<string>();
+    [...node.implicits].forEach((implicit) => this.liveVars.add(implicit));
+    // node.vReturnParameters.vParameters.forEach((p) => this.liveVars.add(p.name));
+    // analyse the block
+    if (node.vBody) {
+      this.dispatchVisit(node.vBody, ast);
+    }
+    // remove the parameters and implicits
+    [...node.implicits].forEach((implicit) => {
+      if (!this.liveVars.delete(implicit)) {
+        throw new TranspileFailedError(
+          `Implicit variable dead. ${implicit} in function ${node.name}.`,
+        );
+      }
+    });
+    node.vParameters.vParameters.forEach((p) => {
+      if (!this.liveVars.delete(p.name) && node.functionStubKind === FunctionStubKind.None) {
+        node.functionStubKind;
+        // console.log(`Dead parameter found ${p.name} in function ${node.name}`);
+      }
+    });
+    this.liveVars = new Set<string>();
+  }
+
+  visitIdentifier(node: Identifier, _ast: AST): void {
+    if (
+      node.vReferencedDeclaration &&
+      node.vReferencedDeclaration.getClosestParentByType(CairoFunctionDefinition)
+    ) {
+      this.liveVars.add(node.name);
+    }
+  }
+
+  visitBlock(node: Block, ast: AST): void {
+    node.vStatements
+      .slice(0)
+      .reverse()
+      .map((child) => this.dispatchVisit(child, ast));
+  }
+
+  visitUncheckedBlock(node: UncheckedBlock, ast: AST): void {
+    node.vStatements
+      .slice(0)
+      .reverse()
+      .map((child) => this.dispatchVisit(child, ast));
+  }
+
+  visitVariableDeclarationStatement(node: VariableDeclarationStatement, ast: AST): void {
+    node.vDeclarations.forEach((declaration) => this.liveVars.delete(declaration.name));
+    if (node.vInitialValue) {
+      this.dispatchVisit(node.vInitialValue, ast);
+    }
+  }
+
+  visitFunctionCall(node: FunctionCall, ast: AST): void {
+    // ignore the function identifier
+    for (const arg of node.vArguments) {
+      this.dispatchVisit(arg, ast);
+    }
+  }
+
+  visitIfStatement(node: IfStatement, ast: AST): void {
+    ensureBothBranchesAreBlocks(node, ast);
+
+    const liveVarsBefore = new Set(this.liveVars);
+    this.dispatchVisit(node.vTrueBody, ast);
+    const liveVarsAfterTrueBody = new Set(this.liveVars);
+    this.liveVars = new Set(liveVarsBefore);
+    let liveVarsAfterFalseBody = new Set<string>();
+    if (node.vFalseBody) {
+      this.dispatchVisit(node.vFalseBody, ast);
+      liveVarsAfterFalseBody = new Set(this.liveVars);
+    }
+    this.liveVars = new Set(liveVarsBefore);
+    this.dispatchVisit(node.vCondition, ast);
+    this.liveVars = new Set([
+      ...this.liveVars,
+      ...liveVarsAfterFalseBody,
+      ...liveVarsAfterTrueBody,
+    ]);
```

Comment:
I think this last assignation to `liveVars` is unnecesary, no further node visits are done after this point

Reply:
This is a class-level record of the `liveVars` at a point in the code. The visited ifStatement might be in a block with statements which proceed it. These statements will not have been visited yet. The live vars of an if is the union of the live vars of the condition and the two branches. This line is correct.